home/utils.py

```diff
@@ -600,26 +655,63 @@ def from_platform_body_element(
         return message
 
 
+@dataclass
+class VariationMessage:
+    body: str = ""
+    title: str = ""
+
+    @classmethod
+    def from_variation(cls, variation: blocks.StreamValue.StreamChild):
+        message = cls()
+        message.body = variation.get("message")
+        profile_fields = []
+        for profile_field in variation.get("variation_restrictions").raw_data:
+            profile_fields.append(f"{profile_field['type']}: {profile_field['value']}")
+        message.title = ", ".join(profile_fields)
+        return message
+
+
+@dataclass
+class VariationMessageList:
+    variations: Tuple[VariationMessage]
```

Comment:
You've specified this as a tuple type, but everywhere I see it being used, it's being used with a list, so maybe it should be `List` instead?